hydrus/resources.py

```diff
@@ -238,42 +227,39 @@ def get(self, path: str) -> Response:
         if isinstance(auth_response, Response):
             return auth_response
         endpoint_ = checkEndpoint("GET", path)
-        if endpoint_['method']:
+        if not endpoint_['method']:
+            # If endpoint and Get method not supported in the API
+            abort(endpoint_['status'])
+        elif path in get_doc().collections:
             # If endpoint and GET method is supported in the API
-            if path in get_doc().collections:
-                # If collection name in document's collections
-                collection = get_doc().collections[path]["collection"]
-                try:
-                    # Get collection details from the database
-                    response = crud.get_collection(
-                        get_api_name(), collection.class_.title, session=get_session(), path=path)
-                    return set_response_headers(
-                        jsonify(hydrafy(response, path=path)))
-
-                except ClassNotFound as e:
-                    status_code, message = e.get_HTTP()
-                    return set_response_headers(
-                        jsonify(message), status_code=status_code)
+            # and collection name in document's collections
+            collection = get_doc().collections[path]["collection"]
+            try:
+                # Get collection details from the database
+                response = crud.get_collection(
+                    get_api_name(), collection.class_.title, session=get_session(), path=path)
+                return set_response_headers(jsonify(hydrafy(response, path=path)))
 
-            # If class is supported
-            elif path in get_doc().parsed_classes and "{}Collection".format(path) not in get_doc(
-            ).collections:
-                try:
-                    class_type = get_doc().parsed_classes[path]['class'].title
-                    response = crud.get_single(
-                        class_type,
-                        api_name=get_api_name(),
-                        session=get_session(),
-                        path=path)
-                    return set_response_headers(
-                        jsonify(hydrafy(response, path=path)))
+            except ClassNotFound as e:
+                status_code, message = e.get_HTTP()
+                return set_response_headers(jsonify(message), status_code=status_code)
 
-                except (ClassNotFound, InstanceNotFound) as e:
-                    status_code, message = e.get_HTTP()
-                    return set_response_headers(
-                        jsonify(message), status_code=status_code)
+        # If endpoint and GET method is supported in the API and class is supported
+        elif path in get_doc().parsed_classes and "{}Collection".format(
```

Comment:
Does this pass PEP8?

Reply:
Also checking the same condition twice, maybe store in a variable and use? That is why there was nested block, you reduced nesting but there are more calls to the session object. Nesting is okay in favour of speed, more calls to session makes it slower.

Reply:
`Does this pass PEP8?` I know it seems weird, but it does comply :sweat_smile: